pkgs/top-level/all-packages.nix

```diff
@@ -16882,6 +16882,10 @@ with pkgs;
 
   minikube = callPackage ../applications/networking/cluster/minikube {
     inherit (darwin.apple_sdk.frameworks) vmnet;
+    drivers = lib.optionals stdenv.isLinux [
```

Comment:
Could you move this in the argument definition in order to reduce the size of `all-packages.nix`?

Reply:
How would I do this cleanly? This comes to mind:

```nix
{ stdenv, buildGoPackage, fetchFromGitHub, fetchurl, go-bindata, libvirt, qemu
, gpgme, makeWrapper, hostPlatform, vmnet, python, pkgconfig
, pkgs, drivers ? stdenv.lib.optionals stdenv.isLinux [ pkgs.docker-machine-kvm pkgs.docker-machine-kvm2 ]
}:
```

But I don't like adding a `pkgs` argument. Or do I do something like:

```nix
{ stdenv, buildGoPackage, fetchFromGitHub, fetchurl, go-bindata, libvirt, qemu
, gpgme, makeWrapper, hostPlatform, vmnet, python, pkgconfig
, kvmDriver ? stdenv.isLinux, docker-machine-kvm
, kvm2Driver ? stdenv.isLinux, docker-machine-kvm2
}:
```

and handle those specifically in `binPath`?

Reply:
Why we could not just add `docker-machine-kvm2` to the argument list as it is done for `docker-machine-kvm`? Moreover, `docker-machine-kvm` is supposed to be removed so it will remain only one driver.

Reply:
I agree with @nlewo: Just add that package as you do now (with `docker-machine-kvm`) and internally create a list of supported drivers.

```nix
let
   drivers = filter (d: d != null) ([]
             ++ lib.optionals stdenv.isLinux [ docker-machine-kvm docker-machine-kvm2 ]
             ++ lib.optionals stdenv.isDarwn [ … ]
   );
in …
```
Note: I am not sure if the `null` filter is required here. Thought is that it will allow people to set inputs to `null` in their overrides and remove specific drivers if they do not want them to be available.

If you want to allow people to add packages (as drivers) that aren't specified within the `minikube` derivation: Add an extraDrivers argument that defaults to `[]` and add that to the list of drivers that are being assembled in the packages expression.